### ISSUE TITLE
[Bug]: trim null in Document/Page 

### ISSUE THREAD

**Issue by ghost** [pimcore/pimcore#12004]
### Expected behavior

Service can perform Service::copyRecursive from Document/Page

### Actual behavior

Crash on perform Service::copyRecursive from Document/Page

Service.php#164
```
        if (method_exists($new, 'setPrettyUrl')) {
            $new->setPrettyUrl(null);
        }
```

Page.php#174
```
    /**
     * @param string $prettyUrl
     *
     * @return $this
     */
    public function setPrettyUrl($prettyUrl)
    {
        $this->prettyUrl = '/' . trim($prettyUrl, ' /');
        if (strlen($this->prettyUrl) < 2) {
            $this->prettyUrl = null;
        }

        return $this;
    }
```

and trim can not handle null values, also phpdoc say it can not receive null as prettyUrl.

### Steps to reproduce

We had this issue, by recreating database for testing szenarios.

Fix?: 
```
/**
 * @param string|null $prettyUrl
 *
 * @return $this
 */
public function setPrettyUrl($prettyUrl)
{
    if (null == $prettyUrl) {
        $this->prettyUrl = null;
        return $this;
    }
    $this->prettyUrl = '/' . trim($prettyUrl, ' /');
    if (strlen($this->prettyUrl) < 2) {
        $this->prettyUrl = null;
    }
    return $this;
}
```

**Comment by ghost** [comment]
It looks like the Dao in Testszenario will also trigger this issue.

Affected
* prettyUrl
* targetGroupIds

```
#0 /var/www/html/vendor/pimcore/pimcore/lib/Model/AbstractModel.php(205): Pimcore\Model\Document\Page->setTargetGroupIds(NULL)\n
#1 /var/www/html/vendor/pimcore/pimcore/lib/Model/AbstractModel.php(187): Pimcore\Model\AbstractModel->setValue('targetGroupIds', NULL)\n
#2 /var/www/html/vendor/pimcore/pimcore/lib/Model/Dao/DaoTrait.php(45): Pimcore\Model\AbstractModel->setValues(Array)\n

```

**Comment by blankse** [comment]
@galdp I created 2 PRs for this. Can you test it?

**Comment by dvesh3** [comment]
Fixed by #12008 & #12009

**Comment by ghost** [comment]
> @galdp I created 2 PRs for this. Can you test it?

@blankse Tests passing, thanks
